### DISCUSSION TITLE
Cart returns an internal server error if it contains any archived or deleted products in cart lines

### DISCUSSION THREAD

**Discussion by yuzushioh** [Shopify/storefront-api-feedback#94] [Ideas & Requests]
This [query](https://shopify.dev/api/storefront/2022-01/objects/Cart#connection-cart-lines) returns an internal server error when cart lines contain any products that have been deleted or archived by merchants. The cart object should handle it internally by removing those products or adding a field that indicates the products statuses and should not return internal errors since it's not providing any information to storefronts for what's going on.

**Comment by johncraigcole** [comment]
Thanks for sharing this @yuzushioh I've gone ahead and created an issue.

**Reply by yuzushioh** [reply]
Thank you so much for doing that 🙏🏻

**Reply by yuzushioh** [reply]
FYI, cart object returns also an internal error when it contains a product which has been removed from the sales channel.

**Comment by pieperz** [comment]
I'm sure i'm not the only one still facing this bug so I thought I would post this video with an explanation of my findings and how to work around this bug. 

https://d.pr/v/hqE9lC

If any one has another workaround or if this does not work for your headless site I'd love to know.

**Reply by yuzushioh** [reply]
Shopify should fix this bug right now. It's been here for so long :(
